tests/TestCase/Database/FunctionsBuilderTest.php

```diff
@@ -151,4 +151,48 @@ public function testNow()
         $this->assertInstanceOf('Cake\Database\Expression\FunctionExpression', $function);
         $this->assertEquals("CURRENT_TIME()", $function->sql(new ValueBinder));
     }
+
+    /**
+     * Tests generating a EXTRACT() function
+     *
+     * @return void
+     */
+    public function testExtract()
+    {
+        $function = $this->functions->extract('day', 'created');
+        $this->assertInstanceOf('Cake\Database\Expression\FunctionExpression', $function);
+        $this->assertEquals("EXTRACT(day FROM created)", $function->sql(new ValueBinder));
+
+        $function = $this->functions->datePart('year', 'modified');
+        $this->assertInstanceOf('Cake\Database\Expression\FunctionExpression', $function);
+        $this->assertEquals("EXTRACT(year FROM modified)", $function->sql(new ValueBinder));
+    }
+
+    /**
+     * Tests generating a DATE_ADD() function
+     *
+     * @return void
+     */
+    public function testDateAdd()
+    {
+        $function = $this->functions->dateAdd('created', -3, 'day');
+        $this->assertInstanceOf('Cake\Database\Expression\FunctionExpression', $function);
+        $this->assertEquals("DATE_ADD(created, INTERVAL -3 day)", $function->sql(new ValueBinder));
+    }
+
+    /**
+     * Tests generating a DAYOFWEEK() function
+     *
+     * @return void
+     */
+    public function testDayOfWeek()
+    {
+        $function = $this->functions->dayOfWeek('created');
+        $this->assertInstanceOf('Cake\Database\Expression\FunctionExpression', $function);
+        $this->assertEquals("DAYOFWEEK(created)", $function->sql(new ValueBinder));
+
+        $function = $this->functions->weekday('created');
+        $this->assertInstanceOf('Cake\Database\Expression\FunctionExpression', $function);
+        $this->assertEquals("DAYOFWEEK(created)", $function->sql(new ValueBinder));
+    }
```

Comment:
How will these tests pass against all the drivers? You might need tonhave separate tests for these in each of the driver/dialect tests instead.